pkgs/development/python-modules/fastdtw/default.nix

```
@@ -1,27 +1,38 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , cython
 , numpy
+  # Check Inputs
+, pytest
+, pytestrunner
 }:
 
 buildPythonPackage rec {
   pname = "fastdtw";
   version = "0.3.4";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "2350fa6ec36bcad186eaf81f46eff35181baf04e324f522de8aeb43d0243f64f";
+  src = fetchFromGitHub {
+    owner = "slaypni";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0irc5x4ahfp7f7q4ic97qa898s2awi0vdjznahxrfjirn8b157dw";
   };
 
   nativeBuildInputs = [
     cython
+  ];
+
+  propagatedBuildInputs = [
     numpy
   ];
 
-  # Pypi's tarball doesn't contain tests
-  doCheck = false;
   pythonImportsCheck = [ "fastdtw.fastdtw" ];
+  checkInputs = [ pytest pytestrunner ];
+  preCheck = ''
+    cd /build/source
```

this just puts you at the root of the src directory, instead move into the tests directory
```suggestion
    cd tests # avoid importing from local files
```